src/main/java/org/embulk/input/marketo/MarketoServiceImpl.java

```diff
@@ -0,0 +1,155 @@
+
+    public MarketoServiceImpl(MarketoRestClient marketoRestClient)
+    {
+        this.marketoRestClient = marketoRestClient;
+    }
+
+    @Override
+    public File extractLead(Date startTime, Date endTime, List<String> extractedFields, int pollingTimeIntervalSecond, int bulkJobTimeoutSecond)
+    {
+        String exportID = marketoRestClient.createLeadBulkExtract(startTime, endTime, extractedFields);
+        marketoRestClient.startLeadBulkExtract(exportID);
+        try {
+            marketoRestClient.waitLeadExportJobComplete(exportID, pollingTimeIntervalSecond, bulkJobTimeoutSecond);
+        }
+        catch (InterruptedException e) {
+            LOGGER.error("Exception when waiting for export job id: {}", exportID, e);
+            throw new DataException("Error when wait for bulk extract");
+        }
+        InputStream leadBulkExtractResult = marketoRestClient.getLeadBulkExtractResult(exportID);
+        return saveExtractedFile(exportID, leadBulkExtractResult);
+    }
+
+    private File saveExtractedFile(String exportID, InputStream leadBulkExtractResult)
+    {
+        File tempFile = Exec.getTempFileSpace().createTempFile(DEFAULT_FILE_FORMAT);
+        try (OutputStream fileOuputStream = new FileOutputStream(tempFile)) {
+            ByteStreams.copy(leadBulkExtractResult, fileOuputStream);
+        }
+        catch (IOException e) {
+            LOGGER.error("Encounter exception when download bulk extract file, job id [{}]", exportID, e);
+            throw new DataException("Can't download bulk extract file");
+        }
+        return tempFile;
+    }
+
+    @Override
+    public File extractAllActivity(Date startTime, Date endTime, int pollingTimeIntervalSecond, int bulkJobTimeoutSecond)
+    {
+        String exportID = marketoRestClient.createActitvityExtract(startTime, endTime);
+        marketoRestClient.startActitvityBulkExtract(exportID);
+        try {
+            marketoRestClient.waitActitvityExportJobComplete(exportID, pollingTimeIntervalSecond, bulkJobTimeoutSecond);
+        }
+        catch (InterruptedException e) {
+            LOGGER.error("Exception when waiting for export job id: {}", exportID, e);
+            throw new DataException("Error when wait for bulk extract");
+        }
+        InputStream activitiesBulkExtractResult = marketoRestClient.getActivitiesBulkExtractResult(exportID);
+        return saveExtractedFile(exportID, activitiesBulkExtractResult);
+    }
+
+    @Override
+    public Iterable<ObjectNode> getAllListLead(List<String> fieldNames)
+    {
+        RecordPagingIterable<ObjectNode> lists = marketoRestClient.getLists();
+        List<Iterable<ObjectNode>> iterables = new ArrayList<>();
+        for (ObjectNode node : lists) {
+            final String id = node.get("id").asText();
+            iterables.add(Iterables.transform(marketoRestClient.getLeadsByList(id, fieldNames), new Function<ObjectNode, ObjectNode>()
+            {
+                @Nullable
```

Comment:
Should remove `@Nullable` here as well as in the method arguments because it is accessed `@NotNull`